### PR TITLE
modify(std/bytes): Rename functions, Update docs, Modify bytes.concat, bytes.indexOf, and bytes.lastIndexOf

### PR DESCRIPTION
**Pull request by yos1p**
Issue: #8610 

I'm just going to drop the code here. Feel free to let me know what else we can do to improve

### CODE REVIEW

File: std/bytes/mod.ts [hunk 1]
```diff
@@ -98,36 +98,47 @@ export function repeat(origin: Uint8Array, count: number): Uint8Array {
...
- * @param b array to concatenate with origin
+/** Concatenate multiple binary arrays and return new one.
+ * @param origin binary array to concatenate
+ * @param buf binary arrays to concatenate with origin
  */
-export function concat(origin: Uint8Array, b: Uint8Array): Uint8Array {
-  const output = new Uint8Array(origin.length + b.length);
-  output.set(origin, 0);
-  output.set(b, origin.length);
+export function concat(a: Uint8Array, ...buf: Uint8Array[]): Uint8Array {
```
Comment (bartlomieju) [inline std/bytes/mod.ts:127]: I believe there should be only a single argument to this function (`...buf`)

Comment (yos1p) [reply std/bytes/mod.ts:127]: Sounds good to me as well. I will make the changes.

**Review by wperron — APPROVED** [review]
LGTM - I believe we also discussed whether or not to rename some of the input parameters as well, not something blocking in my opinion though

File: std/bytes/mod.ts [hunk 2]
```diff
@@ -77,7 +77,7 @@ export function hasPrefix(source: Uint8Array, prefix: Uint8Array): boolean {
  * @param source source array
  * @param suffix suffix array to check in source
  */
```
Comment (ry) [inline std/bytes/mod.ts:79]: Can you mention the RangeError in the JSDoc?

File: std/bytes/test.ts [hunk 3]
```diff
@@ -111,6 +111,19 @@ Deno.test("[bytes] concat empty arrays", () => {
...
+  const u2 = encode("W");
+  const u3 = encode("o");
+  const u4 = encode("r");
+  const u5 = encode("l");
+  const u6 = encode("d");
+  const joined = concat(u1, u2, u3, u4, u5, u6);
+  assertEquals(decode(joined), "Hello World");
+  assert(u1 !== joined);
+  assert(u2 !== joined);
+});
```
Comment (ry) [inline std/bytes/test.ts:125]: 👍

**Review by ry — COMMENTED** [review]
Thank you @yos1p

File: std/bytes/test.ts [hunk 4]
```diff
@@ -124,15 +124,15 @@ Deno.test("[bytes] concat multiple arrays", () => {
...
-Deno.test("[bytes] contain", () => {
+Deno.test("[bytes] contains", () => {
   const source = encode("deno.land");
   const pattern = encode("deno");
   assert(contains(source, pattern));
 
   assert(contains(new Uint8Array([0, 1, 2, 3]), new Uint8Array([2, 3])));
 });
 
-Deno.test("[io/tuil] copyBytes", function (): void {
```
Comment (yos1p) [inline std/bytes/test.ts:122]: Anyone know why this is [io/util]? Is it on purpose?

Comment (yos1p) [reply std/bytes/test.ts:122]: I rename it to [bytes] given the preceding functions. But let me know if I should revert this.

File: std/bytes/mod.ts (hunk 2 above)
Comment (yos1p) [reply std/bytes/mod.ts:79]: Yes I can do that

**Comment by yos1p** [comment]
I also added changes for indexOf/lastIndexOf to take start index as optional parameter. The start index for both functions is based on Array.indexOf and Array.lastIndexOf from [javascript reference document](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/indexOf).

I think it will make it easier what will be expected by JS developers. But let me know if that should be changed (e.g throw RangeError instead)

File: std/bytes/test.ts (hunk 4 above)
Comment (bartlomieju) [reply std/bytes/test.ts:122]: That's okay @yos1p; IIRC this method was in `std/io/util.ts` at some point

**Review by ry — APPROVED** [review]
LGTM - thank you @yos1p

**Pull request merged**